src/renate/updaters/learner.py

```diff
@@ -125,53 +141,15 @@ def _create_metrics_collections(
             }
         )
 
-    def state_dict(self, **kwargs) -> Dict[str, Any]:
-        """Returns the state of the learner."""
-        return {
+    def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        learner_state_dict = {
             "learner_class_name": self.__class__.__name__,
```

Comment:
It seems like this is not used anymore. I don't mind leaving it in, but we could remove it. Your call.

Reply:
it isnt. But I left it in as a sanity check.